### DISCUSSION TITLE
Support for reverse proxy subdirectories

### DISCUSSION THREAD

**Discussion by mind-overflow** [standardnotes/forum#2752] [Suggestions]
When the syncing-server has been rewritten from scratch in Node, support for running it in subdirectories has been added. For example, instead of running it on `https://sync.example.org`, you can now run it on `https://services.example.org/notes/sync`.  
This is very simple, and requires no configuration from the server's side - you only have to edit the reverse proxy. Example (using Apache2, but very similar in nginx):
```
# Standard Notes Reverse Proxy
<VirtualHost *:443>
        ServerName services.example.org

        ProxyRequests Off
        ProxyPreserveHost On
        ProxyVia Full
        <Proxy *>
                Require all granted
        </Proxy>

        <Location "/notes/sync">
            ProxyPass "http://127.0.0.1:3000"
            ProxyPassReverse "http://127.0.0.1:3000"
        </Location>
</VirtualHost>
```  
  
This is very useful in case someone is running multiple services from the same subdomain. Some other programs implement it in a different way, for example by providing a `URL_ROOT` variable in the `.env` file, where you can just write `URL_ROOT=/notes/sync` and it will be appended after the host and before any URL. Then, in your reverse proxy, you do this:
```
        <Location "/notes/sync">
            ProxyPass "http://127.0.0.1:3000/notes/sync"
            ProxyPassReverse "http://127.0.0.1:3000/notes/sync"
        </Location>
```
  
And everything works smoothly.  
Unfortunately, this is not the case with the web-ui server for Standard Notes. In fact, if I do like the first example, the page loads but it's blank, because everything is returning 404:    
  
```
GET https://services.example.org/assets/javascripts/app-3e75b90384c56a25ff0f63add542f6822ae22f4ed9e5df7ccf0c385e755ae8e8.js net::ERR_ABORTED 404
```
  
This is, of course, because the web server is not aware of the subdirectory (unlike the syncing server, which seems to automatically detect or adapt to it).   
The correct URL would be:
```
GET https://services.example.org/notes/web/assets/javascripts/app-3e75b90384c56a25ff0f63add542f6822ae22f4ed9e5df7ccf0c385e755ae8e8.js
```

Note the added `/notes/web` in the url.  
If I try the second way, the page itself returns 404 because of course there is no page available at `/notes/web/` in the web-ui standard notes server itself, which is where the `ProxyPass` directives points to.  
  
So, I'm asking to kindly fix this, either like the syncing-server, or by doing what other programs are doing and appending the base-url before the real url (depending on which one is easier to implement). An example that comes to mind is remark42.  
  
Of course I'm just suggesting it, and not requiring it - you're doing this for free and it's very remarkable. I just thought it would help people in a similar scenario.  
  
Thank you.

**Comment by JaspalSuri** [comment]
Hi @mind-overflow, sorry to hear that. I'll forward your feedback to the dev team for consideration.
